js/src/common/states/AlertManagerState.ts

```diff
@@ -1,30 +1,48 @@
-import Alert from '../components/Alert';
+import Mithril from 'mithril';
+import Alert, { AlertAttrs } from '../components/Alert';
+
+/**
+ * Returned by `AlertManagerState.show`. Used to dismiss alerts.
+ */
+export type AlertIdentifier = number;
+
+export interface AlertState {
+  componentClass: typeof Alert;
+  attrs: AlertAttrs;
+  children: Mithril.Children;
+}
 
 export default class AlertManagerState {
-  constructor() {
-    this.activeAlerts = {};
-    this.alertId = 0;
-  }
+  protected activeAlerts: { [ids: string]: AlertState } = {};
```

Comment:
Shouldn't the `ids` be of type `number`, or better `AlertIdentifier`?

Reply:
Also, `ids` here should be singular - you're typing each individual object key as one, not combined. Not that it matters, but you know 😛

Reply:
According to TS docs, it has to be either number or string. I guess number is better here